File: c/adler32.c
#include <stdio.h>

int main() {
    int MOD_ADLER = 65521;
    int a = 1;
    int b = 0;
    for (unsigned int i = 0; i < 250000000; i++) {
        a = (a + i) % MOD_ADLER;
        b = (b + a) % MOD_ADLER;
    } 
    unsigned int result = b << 16 | a;
    printf("|c|adler32|%x|N/A|\n", (result));
}

File: nim/adler32.nim
import std/monotimes
import times
import strutils

proc main() =
    let start = getMonoTime()
    const mod_adler: int32 = 65521
    var a: int32 = 1
    var b: int32 = 0
    for i in 0i32..<250000000:
        a = (a + i) %% mod_adler
        b = (b + a) %% mod_adler
    let result = uint32(b) * uint32(65536) + uint32(a)

    let delta = (getMonoTime() - start).inMilliseconds()
    echo "|nim|adler32|", result.toHex, "|", delta, "|"

when isMainModule:
    main()

